### PR TITLE
Adding SafeCast variants for signed integers

### LINKED ISSUE #2240: Add SafeCast variants for signed integers
[`SafeCast`](https://docs.openzeppelin.com/contracts/3.x/api/utils#SafeCast) currently has support for downcasting uints and casting them to int256, but there's no safe way to downcast signed integers.

This is slightly trickier because the two's complement integer representation has different ranges for positive and negative numbers, but should be easy to do anyway.

### PR DESCRIPTION
**Pull request by julianmrodri**
#2240 

Fixes #2240 

**Description:** * Added `SafeCast` variants for signed Integers. This includes the following functions:

- `toInt8(int256 x)`: Downcasts to int8 from int256
- `toInt16(int256 x)`: Downcasts to int16 from int256
- `toInt32(int256 x)`: Downcasts to int32 from int256
- `toInt64(int256 x)`: Downcasts to int64 from int256
- `toInt128(int256 x)`: Downcasts to int128 from int256

Also added the corresponding Tests and adjustments to the Mockup contract. All tests works correctly and linter is also validated.

**Modified Files:**  

1. contracts/utils/SafeCast.sol
2. contracks/mocks/SafeCastMock.sol
3. test/utils/SafeCast.test.js

### CODE REVIEW

**Comment by julianmrodri** [comment]
@nventuro @frangio  Ready to be reviewed. Let me know in case any chances or modifications need to be done. Thanks!

**Comment by julianmrodri** [comment]
@nventuro @frangio  Not sure why Tests are failing all running fine on my side. Is it possible they are not related to the SafeCast changes?

image

File: contracts/utils/SafeCast.sol [hunk 1]
```diff
@@ -4,19 +4,19 @@ pragma solidity ^0.6.0;
...
  * assume that overflows raise errors. `SafeCast` restores this intuition by
  * reverting the transaction when such an operation overflows.
  *
  * Using this library instead of the unchecked operations eliminates an entire
  * class of bugs, so it's recommended to use it always.
  *
- * Can be combined with {SafeMath} to extend it to smaller types, by performing
- * all math on `uint256` and then downcasting.
+ * Can be combined with {SafeMath} and {SignedSafeMath} to extend it to smaller types, by performing
+ * all math on `uint256/int256` and then downcasting.
```
Comment (nventuro) [inline contracts/utils/SafeCast.sol:19]:
```suggestion
 * all math on `uint256` and `int256` and then downcasting.
```

**Comment by julianmrodri** [comment]
Thanks @nventuro! I was sure I had to update the **CHANGELOG** as well but it looked quite sensitive as I wasn't sure I should include it under the next release, so I left it you to you guys.

I don't have any concerns about these functions I did quite a bunch of manual testing as well interacting with the Contract and seems to work as expected. The formula extrapolated nicely to all the different types. 

Let me know if there's anything else to do or include and I can take care. Thanks again!

**Comment by nventuro** [comment]
Awesome, thanks! We don't ask contributors to edit the changelog to reduce the burden on them, specially since it's very simple for us to do from the web UI.

Thank you very much once more!

**Pull request merged**